.clj-kondo/hooks/common.clj

```diff
@@ -200,6 +200,34 @@
                 body))]
     {:node node*}))
 
+(defn- let-second-inner [body bindings]
+  (let [binding-infos (for [[model {[binding value] :children}] (partition 2 bindings)]
+                        {:model   model
+                         :binding binding
+                         :value   (or value
+                                      (hooks/token-node 'nil))})]
+    (-> (hooks/vector-node
+         [(hooks/vector-node (map :model binding-infos))
+          (-> (hooks/list-node (list* (hooks/token-node `let)
+                                    (hooks/vector-node (mapcat (juxt :binding :value) binding-infos))
+                                    body))
+              (with-meta (meta body)))])
+        (with-meta (meta body)))))
+
+(defn let-second
+  "Helper for macros that have a shape like, where the let is for second arg.
+
+    (my-macro x y
+    ...)
+
+    =>
+
+    (let [y nil]
+    ...)"
+  [{:keys [node]}]
+  (let [[_ db-ref binding+opts & body] (:children node)]
+    {:node (let-second-inner body [db-ref binding+opts])}))
```

Comment:
This is a general hook, not specific to the fixture. Should `db-ref` be renamed?

Reply:
The `db-ref` binding also appears unused if it's not otherwise used. Your tests don't trigger this warning, so maybe you could ignore it.

If I do something like this
```
    (t2.with-temp/with-temp [Dashboard dashboard {:name "Test Dashboard"}]
      (with-link-card-fixture-for-dashboard dashboard [...] ...)
```
And don't use dashboard anywhere else other than passing it as an arg to `with-link-card-fixture-for-dashboard`, then it will trigger a clj-kondo warning:

`test/metabase/dashboard_subscription_test.clj:640:11: warning: unused binding dashboard`

I think you'll need to do something like this to trick kondo into thinking it's used https://github.com/metabase/metabase/blob/15e9a71fb0c3f4f888dcaeb66f0b4efeda013d07/.clj-kondo/hooks/common.clj#L272